<a href="https://colab.research.google.com/github/muhamadrefo/ImageClassificationColab/blob/master/NaiveBayesClassifier_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Prepare Environment

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [4]:
#Importing Packages

import matplotlib.pyplot as plt
import csv
import numpy as np
import pandas as pd
from sklearn import model_selection
from sklearn.model_selection import train_test_split as tt
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score

%matplotlib inline

In [5]:
#Filtered Dataset
%cd /content/drive/My Drive/TestingPlace/PlaceofDatabase/CNN/
#Unfiltered Dataset
#%cd /content/drive/My Drive/Batik/

/content/drive/My Drive/TestingPlace/PlaceofDatabase/CNN


#Reading dataset

In [23]:
#col_names = ["Path", "Dissimilarity", "Contrast", "Homogeneity", "Energy", "Correlation", "ASM" , "Label"]
#datatrain = pd.read_csv('DataTraining.csv', names=col_names)

data_train = pd.read_csv('DataTraining.csv') #DataTraining.csv nor DataValidation.csv
#data_train = pd.read_csv('DataValidation.csv')
pd.set_option('display.max_columns', 100)
data_train.head(15) #display first 15 data

Path  Dissimilarity  Contrast  \
0   dataset_Cap_vs_Tulis/train/cap/YCNSBT.39.jpg       0.627122  0.905349   
1   dataset_Cap_vs_Tulis/train/cap/YCNSBT.35.jpg       0.537396  0.718049   
2   dataset_Cap_vs_Tulis/train/cap/YCNSBT.36.jpg       0.594090  0.842489   
3   dataset_Cap_vs_Tulis/train/cap/YCNSBT.37.jpg       0.609385  0.836083   
4   dataset_Cap_vs_Tulis/train/cap/YCNSBT.34.jpg       0.601197  0.844070   
5   dataset_Cap_vs_Tulis/train/cap/YCNSBT.32.jpg       0.656130  0.969711   
6   dataset_Cap_vs_Tulis/train/cap/YCNSBT.33.jpg       0.557755  0.748859   
7   dataset_Cap_vs_Tulis/train/cap/YCNSBT.24.jpg       0.657111  0.960802   
8   dataset_Cap_vs_Tulis/train/cap/YCNSBT.25.jpg       0.597874  0.836263   
9   dataset_Cap_vs_Tulis/train/cap/YCNSBT.28.jpg       0.548847  0.727018   
10  dataset_Cap_vs_Tulis/train/cap/YCNSBT.26.jpg       0.550969  0.743234   
11   dataset_Cap_vs_Tulis/train/cap/YCNSBT.3.jpg       0.498058  0.698731   
12  dataset_Cap_vs_Tulis/train/cap/YCNSBT.23.jpg       0.598795  0.844671   
13  dataset_Cap_vs_Tulis/train/cap/YCNSBT.21.jpg       0.614890  0.864450   
14   dataset_Cap_vs_Tulis/train/cap/YCNSBT.2.jpg       0.510130  0.721092   

    Homogeneity    Energy  Correlation       ASM Label  
0      0.714101  0.363739     0.449124  0.132306   cap  
1      0.749297  0.381303     0.548324  0.145392   cap  
2      0.727348  0.379189     0.437346  0.143784   cap  
3      0.717911  0.383743     0.414907  0.147259   cap  
4      0.723463  0.364463     0.548681  0.132833   cap  
5      0.703073  0.363779     0.436385  0.132335   cap  
6      0.739961  0.391705     0.455864  0.153433   cap  
7      0.701605  0.373753     0.384801  0.139691   cap  
8      0.724778  0.370439     0.472178  0.137225   cap  
9      0.743285  0.385256     0.536179  0.148422   cap  
10     0.743635  0.380415     0.534508  0.144715   cap  
11     0.770924  0.496100     0.661364  0.246115   cap  
12     0.724906  0.374617     0.446828  0.140338   cap  
13     0.717309  0.375527     0.434121  0.141021   cap  
14     0.765693  0.517909     0.387049  0.268230   cap

In [24]:
# show columns
data_train.columns 

Index(['Path', 'Dissimilarity', 'Contrast', 'Homogeneity', 'Energy',
       'Correlation', 'ASM', 'Label'],
      dtype='object')

In [25]:
data_train

Path  Dissimilarity  Contrast  \
0      dataset_Cap_vs_Tulis/train/cap/YCNSBT.39.jpg       0.627122  0.905349   
1      dataset_Cap_vs_Tulis/train/cap/YCNSBT.35.jpg       0.537396  0.718049   
2      dataset_Cap_vs_Tulis/train/cap/YCNSBT.36.jpg       0.594090  0.842489   
3      dataset_Cap_vs_Tulis/train/cap/YCNSBT.37.jpg       0.609385  0.836083   
4      dataset_Cap_vs_Tulis/train/cap/YCNSBT.34.jpg       0.601197  0.844070   
..                                              ...            ...       ...   
420  dataset_Cap_vs_Tulis/train/tulis/YTNSWT.10.jpg       0.420203  0.572349   
421  dataset_Cap_vs_Tulis/train/tulis/YTNSWT.11.jpg       0.557255  0.886971   
422  dataset_Cap_vs_Tulis/train/tulis/YTNSWT.12.jpg       0.484585  0.694867   
423  dataset_Cap_vs_Tulis/train/tulis/YTNSWT.13.jpg       0.428832  0.573971   
424   dataset_Cap_vs_Tulis/train/tulis/YTNSWT.1.jpg       0.399123  0.533933   

     Homogeneity    Energy  Correlation       ASM  Label  
0       0.714101  0.363739     0.449124  0.132306    cap  
1       0.749297  0.381303     0.548324  0.145392    cap  
2       0.727348  0.379189     0.437346  0.143784    cap  
3       0.717911  0.383743     0.414907  0.147259    cap  
4       0.723463  0.364463     0.548681  0.132833    cap  
..           ...       ...          ...       ...    ...  
420     0.805099  0.529145     0.406434  0.279995  tulis  
421     0.754050  0.461027     0.469459  0.212546  tulis  
422     0.778702  0.504275     0.377737  0.254294  tulis  
423     0.800067  0.515419     0.442994  0.265657  tulis  
424     0.813917  0.540175     0.475306  0.291789  tulis  

[425 rows x 8 columns]

# Encoding the Categorial Data

In [26]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
df = data_train.apply(le.fit_transform)
df.head(15)

Path  Dissimilarity  Contrast  Homogeneity  Energy  Correlation  ASM  \
0     20            299       273          104     118           65  118   
1     17            243       195          154     153           97  153   
2     18            277       247          123     149           60  149   
3     19            288       244          107     159           50  159   
4     16            285       250          114     120           98  120   
5     14            309       294           97     119           59  119   
6     15            256       217          137     175           67  175   
7      9            310       293           91     134           35  134   
8     10            279       245          118     127           73  127   
9     12            249       201          144     163           94  163   
10    11            252       216          145     151           93  151   
11    13            211       189          194     285          201  285   
12     8            280       252          120     138           64  138   
13     7            293       259          106     141           57  141   
14     6            220       199          182     317           38  317   

    Label  
0       0  
1       0  
2       0  
3       0  
4       0  
5       0  
6       0  
7       0  
8       0  
9       0  
10      0  
11      0  
12      0  
13      0  
14      0

# Splitting the Data into Training and Testing Data

In [77]:
X = df.iloc[:, 0:7].values # per array (Path, Diss, Contr, Homog, Energ, Correl, ASM), when start with 0, 'path' included, if start from 1, mean 'path' not included 
Y = df.iloc[:, 7].values # array (Label 'cap' or 'tulis')
X_train, X_test, Y_train, Y_test = tt(X,Y, test_size=0.2, random_state=4)

In [78]:
X_train.shape

(340, 7)

In [79]:
Y.shape

(425,)

In [80]:
X_test.shape

(85, 7)

In [81]:
X.shape

(425, 7)

#Naive Bayes Model Start Here

In [82]:
from sklearn.naive_bayes import GaussianNB
nb = GaussianNB()
nb.fit(X_train,Y_train)
y_pred = nb.predict(X_test)

In [83]:
from sklearn.metrics import classification_report , confusion_matrix , accuracy_score
cm = confusion_matrix(Y_test,y_pred)

In [84]:
print("Confusion Matrix: \n",cm)

Confusion Matrix: 
 [[48  1]
 [ 0 36]]


In [85]:
print("Classification Report: \n", classification_report(Y_test, y_pred))

Classification Report: 
               precision    recall  f1-score   support

           0       1.00      0.98      0.99        49
           1       0.97      1.00      0.99        36

    accuracy                           0.99        85
   macro avg       0.99      0.99      0.99        85
weighted avg       0.99      0.99      0.99        85



In [86]:
print("Accuracy Score: ", accuracy_score(Y_test, y_pred))

Accuracy Score:  0.9882352941176471


# Analytic

using dataset unfiltered
Accuracy Score Training = 0.74
Accuracy Score Validation = 1.0

using dataset filtered
Accuracy Score Training = 0.77
Accuracy Score Validation = 0.55